### ISSUE TITLE
Make FirestoreTemplate extendable

### ISSUE THREAD

**Issue by heroInCommunity** [GoogleCloudPlatform/spring-cloud-gcp#443]
It would be nice to have method `FirestoreTemplate.findAllDocuments` as protected, so it could be overridden in a custom FirestoreTemplate class.

The use case could be if you need to apply the same filtering condition to all queries, e.g. by attributes 'active', 'deleted', 'tenantId' etc.

At the moment this method is private, so the only way to intervene is to use AspectJ, which does not look like a good solution.

**Comment by meltsufin** [comment]
I'd rather enable the usecase more idiomatically than having you extend the `FirestoreTemplate` based on a non-public API. For example, in Spring Data Spanner, we have [`@Where`](https://googlecloudplatform.github.io/spring-cloud-gcp/2.0.2/reference/html/index.html#declarative-filtering-with-where). Are you looking for something like that?

**Comment by heroInCommunity** [comment]
Yes, that would work as well, but SpEL is needed inside of it then.
